File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string s) {
        int ans = 0;
        string str = "";
        for(int i = 0; i < sentence.size(); i++){
            if(str == s){
                return ans+1;
            }
            if(sentence[i] == ' '){
                str = "";
                ans++;
            }
            else str += sentence[i];
        }
        if(str == s) return ans+1;
        return -1;
    }
};

File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.go
func isPrefixOfWord(sentence string, searchWord string) int {
    ans := 0
    str := ""
    for i := 0; i < len(sentence); i++{
        if searchWord == str{
            return ans+1
        }
        if sentence[i] == ' '{
            str = ""
            ans++
        }else{
            str += string(sentence[i])
        }
    }
    if searchWord == str {
        return ans+1
    }
    return -1
}